### DISCUSSION TITLE
[Dev Tools] <RPC Nodes Management Tools>

### DISCUSSION THREAD

**Discussion by JoGetBlock** [zkSync-Community-Hub/zksync-developers#723] [Show and Tell]
### Environment

Mainnet

### Select the Dev Tool you are using

Local Node

### Provide a brief description of the functionality you're trying to implement and the issue you are running into.

When running an RPC node it's crucial to be always aware of the consistency and availability of your node. To do so, you have to utilize some robust management and monitoring tools. Here’s an example of the tools GetBlock is using: 
- Prometheus open-source monitoring system 
- Grafana observability platform; (the latter sources data from the first one.)
- The health sidecar
- Alertmanager service in Slack
- Loadservice
- Auto-switching system

Prometheus gathers metrics and databases to display in Grafana. Prometheus is also bonded to Alertmanager service to inform the team in Slack about all events regarding infrastructure status. The health sidecar helps GetBlock monitor the current height and health of the nodes. To get immediate notifications from the monitoring tool GetBlock connected it to the Alertmanager service in Slack. It helps to get the fastest notification if some issue occurs and always double-check when it’s resolved. The health sidecar is also connected to the auto-switching system. So if the block deviation occurs, the unhealthy node is instantly switched to a healthy one. The last but not the least important thing is to keep the node updated to the latest versions. This way GetBlock constantly monitors blockchains’ GitHub repositories and social medias to find out about the upcoming updates first in hand. 

### Additional Details

If you don’t wanna experience all the hustles associated with running and maintaining your ZKsync RPC node. You can simply connect to RPC nodes for 50+ blockchains with GetBlock.io

**Comment by alhenaai** [comment]
For managing and monitoring your local ZKsync RPC node effectively, you can utilize a similar setup to what GetBlock uses. Here are some tools and their functions that can help ensure the consistency and availability of your node:

- **Prometheus**: This open-source monitoring system collects metrics and databases, which are then displayed in Grafana. It's also connected to the Alertmanager service to inform the team about infrastructure status via Slack.
- **Grafana**: An observability platform that sources data from Prometheus to provide visualizations of the metrics.
- **Health Sidecar**: Monitors the current height and health of the nodes.
- **Alertmanager**: Integrated with Slack for immediate notifications regarding infrastructure issues, ensuring quick awareness and response.
- **Loadservice and Auto-Switching System**: These tools help manage node traffic and automatically switch to a healthy node if any block deviation occurs.

These tools collectively help in maintaining high availability and consistency of your RPC node. If managing your own node seems complex, consider using services like [GetBlock](https://getblock.io/nodes/zksync/) which offers managed RPC node connections for ZKsync and other blockchains.

Your feedback helps us serve you better. If this fully addresses your question, please give it a thumbs up. If it doesn't, please give it a thumbs down. This will inform the community and team so we can improve our support.

**Comment by dutterbutter** [comment]
Thanks for posting, I am going to close the discussion but it will be visible to interested parties.
